test/cql-pytest/test_native_transport.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# The cql-pytest framework is about testing CQL functionality, so
+# implementation details like sstables cannot be tested directly. However,
+# we are still able to reproduce some bugs by tricks such as writing some
+# data to the table and then force it to be written to the disk (nodetool
+# flush) and then trying to read the data again, knowing it must come from
+# the disk.
+#############################################################################
+
+import pytest
+import time
+import nodetool
+from util import new_test_table, new_cql
+from cassandra.cluster import NoHostAvailable
+
+def test_enable_disable_binary(cql, test_keyspace):
+    schema = 'k text, v text, primary key (k)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"INSERT INTO {table} (k,v) VALUES ('foo', 'bar')")
+        cql.execute(f"SELECT * FROM {table}")
+
+        nodetool.disablebinary(cql)
+        select_success = False
+        try:
+            cql.execute(f"SELECT * FROM {table}")
+            select_success = True
+        except NoHostAvailable:
+            pass
+        assert not select_success
+
+        nodetool.enablebinary(cql)
+        select_success = False
+        for i in range(5):
+            try:
+                with new_cql(cql) as ncql:
```

Comment:
The fact you used new_cql here and the change you did to the test.py framework suggests to me a serious problem with this test: It kills the "cql" fixture. Everything is fine if you have a test file with a single test and you run it alone (this is what test.py does), but what if there is another test in this file that needs the "cql" fixture to work? What if you run test/cql-pytest/run and it runs all test files in one pytest (and not a different pytest invocation per test)?

I think you need to find a way for the "cql" fixture to somehow recover after enablebinary, if it's not automatic. Then, this test should use the recovered "cql" like other tests, and shouldn't need to use new_cql.

Reply:
Fair point. Will take a look at that